csharp/src/Google.Protobuf/MessageExtensions.cs

```diff
@@ -54,6 +54,22 @@ public static void MergeFrom(this IMessage message, byte[] data)
         }
 
         /// <summary>
+        /// Merges data from the given byte array slice into an existing message.
+        /// </summary>
+        /// <param name="message">The message to merge the data into.</param>
+        /// <param name="data">The data containing the slice to merge, which must be protobuf-encoded binary data.</param>
+        /// <param name="offset">The offset of the slice to merge.</param>
+        /// <param name="length">The length of the slice to merge.</param>
+        public static void MergeFrom(this IMessage message, byte[] data, int offset, int length)
+        {
+            ProtoPreconditions.CheckNotNull(message, "message");
```

Comment:
At some point I should do a nameof tidy-up... but I understand the point of sticking to the convention of the rest of the file here.